base/range.jl

```diff
@@ -760,8 +739,53 @@ end
+\(x::Number, r::Range)        = range(x\first(r), x\step(r), x\length(r))
+\(x::Number, r::StepRangeLen) = StepRangeLen(x\r.ref, x\r.step, length(r), r.offset)
+\(x::Number, r::LinSpace)     = LinSpace(x \ r.start, x \ r.stop, r.len)
+
+## scalar-range broadcast operations ##
+
+broadcast(::typeof(-), r::OrdinalRange) = range(-first(r), -step(r), length(r))
+broadcast(::typeof(-), r::StepRangeLen) = StepRangeLen(-r.ref, -r.step, length(r), r.offset)
+broadcast(::typeof(-), r::LinSpace) = LinSpace(-r.start, -r.stop, length(r))
 
-/(x::Number, r::Range) = [ x/y for y=r ]
+broadcast(::typeof(+), x::Real, r::AbstractUnitRange) = range(x + first(r), length(r))
+# For #18336 we need to prevent promotion of the step type:
+broadcast(::typeof(+), x::Number, r::AbstractUnitRange) = range(x + first(r), step(r), length(r))
+broadcast(::typeof(+), x::Number, r::Range) = (x+first(r)):step(r):(x+last(r))
+function broadcast(::typeof(+), x::Number, r::StepRangeLen)
+    newref = x + r.ref
+    StepRangeLen{eltype(newref),typeof(newref),typeof(r.step)}(newref, r.step, length(r), r.offset)
+end
+function broadcast(::typeof(+), x::Number, r::LinSpace)
+    LinSpace(x + r.start, x + r.stop, r.len)
+end
+broadcast(::typeof(+), r::Range, x::Number) = broadcast(+, x, r)  # assumes addition is commutative
+
+broadcast(::typeof(-), x::Number, r::Range)      = (x-first(r)):-step(r):(x-last(r))
+broadcast(::typeof(-), x::Number, r::StepRangeLen) = broadcast(+, x, -r)
+function broadcast(::typeof(-), x::Number, r::LinSpace)
+    LinSpace(x - r.start, x - r.stop, r.len)
+end
+
+broadcast(::typeof(-), r::Range, x::Number) = broadcast(+, -x, r)  # assumes addition is commutative
+
+broadcast(::typeof(*), x::Number, r::Range)        = range(x*first(r), x*step(r), length(r))
+broadcast(::typeof(*), x::Number, r::StepRangeLen) = StepRangeLen(x*r.ref, x*r.step, length(r), r.offset)
+broadcast(::typeof(*), x::Number, r::LinSpace)     = LinSpace(x * r.start, x * r.stop, r.len)
+# separate in case of noncommutative multiplication
+broadcast(::typeof(*), r::Range, x::Number)        = range(first(r)*x, step(r)*x, length(r))
+broadcast(::typeof(*), r::StepRangeLen, x::Number) = StepRangeLen(r.ref*x, r.step*x, length(r), r.offset)
+broadcast(::typeof(*), r::LinSpace, x::Number)     = LinSpace(r.start * x, r.stop * x, r.len)
+
+broadcast(::typeof(/), r::Range, x::Number)        = range(first(r)/x, step(r)/x, length(r))
+broadcast(::typeof(/), r::StepRangeLen, x::Number) = StepRangeLen(r.ref/x, r.step/x, length(r), r.offset)
+broadcast(::typeof(/), r::LinSpace, x::Number)     = LinSpace(r.start / x, r.stop / x, r.len)
+# also, separate in case of noncommutative multiplication (division)
+broadcast(::typeof(\), x::Number, r::Range)        = range(x\first(r), x\step(r), x\length(r))
+broadcast(::typeof(\), x::Number, r::StepRangeLen) = StepRangeLen(x\r.ref, x\r.step, length(r), r.offset)
+broadcast(::typeof(\), x::Number, r::LinSpace)     = LinSpace(x \ r.start, x \ r.stop, r.len)
```

Comment:
I would add a deprecation for these methods in favor of the non-dot functions, to help anyone who is expecting them to produce a range.